### PR TITLE
[Local GC] Fix a number of handle table interface violations

### PR DESCRIPTION
**Pull request by swgillespie**
This PR began with some changes that I originally made to build the GC as a standalone library. It has evolved to contain a few things:
  1. Fix usages of `HndGetHandleADIndex` in the VM to use `IGCHandleManager::GetHandleContext`, which already exists on the interface and is used in other places throughout the VM for retrieving the AD index.
  2. Change methods on `IGCHandleManager` that take handle types to refer to the `HandleType` enum as its argument instead of `int`, for the sake of clarity.
  3. Add `HndSetHandleExtraInfo` to the interface (as `SetExtraInfoForHandle`) and change callers to use the new interface method.

### CODE REVIEW

**Comment by swgillespie** [comment]
@dotnet-bot test Windows_NT Checked standalone_gc
@dotnet-bot test Ubuntu Checked standalone_gc

**Comment by swgillespie** [comment]
@dotnet-bot test Tizen armel Cross Release

**Comment by swgillespie** [comment]
@dotnet-bot test Tizen armel Cross Release build

**Comment by swgillespie** [comment]
@adityam @Maoni0 @jkotas PTAL? This should be the last of the handle table interface violations.

File: src/gc/gchandletable.cpp [hunk 1]
```diff
@@ -173,3 +179,11 @@ Object* GCHandleManager::InterlockedCompareExchangeObjectInHandle(OBJECTHANDLE h
 {
     return (Object*)::HndInterlockedCompareExchangeHandle(handle, ObjectToOBJECTREF(object), ObjectToOBJECTREF(comparandObject));
 }
+
+HandleType GCHandleManager::HandleFetchType(OBJECTHANDLE handle)
+{
+    uint32_t type = ::HandleFetchType(handle);
+    assert(type >= 0 && type <= 9);
```
Comment (jkotas) [inline src/gc/gchandletable.cpp]: Maybe nice to use `HNDTYPE_WEAK_WINRT` instead of hardcoded 9

Comment (swgillespie) [inline src/gc/gchandletable.cpp]: will do!

**Comment by swgillespie** [comment]
The Tizen run died with a SIGINT, doesn't sound like it's related to these changes.

@dotnet-bot test Tizen armel Cross Release build

**Comment by swgillespie** [comment]
thanks!

**Pull request merged**